app crash when find xml url backend service is closed.

Caused by: java.lang.RuntimeException: java.net.ConnectException: failed to connect to /192.168.0.254 (port 8080): connect failed: ECONNREFUSED (Connection refused)
                                                                           at com.github.javiersantos.appupdater.RssParser.getInputStream(RssParser.java:58)
                                                                           at com.github.javiersantos.appupdater.RssParser.parse(RssParser.java:32)
                                                                           at com.github.javiersantos.appupdater.UtilsLibrary.getLatestAppVersionXml(UtilsLibrary.java:243)
                                                                           at com.github.javiersantos.appupdater.UtilsAsync$LatestAppVersion.doInBackground(UtilsAsync.java:57)
                                                                           at com.github.javiersantos.appupdater.UtilsAsync$LatestAppVersion.doInBackground(UtilsAsync.java:13)

--- a/library/src/main/java/com/github/javiersantos/appupdater/RssParser.java
+++ b/library/src/main/java/com/github/javiersantos/appupdater/RssParser.java
@@ -11,6 +11,7 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.UnknownHostException;
 
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.parsers.SAXParser;
@@ -51,7 +52,7 @@ class RssParser {
     private InputStream getInputStream() {
         try {
             return rssUrl.openConnection().getInputStream();
-        } catch (FileNotFoundException e) {
+        } catch (FileNotFoundException | UnknownHostException e) {
             Log.e("AppUpdater", "The XML updater file is invalid or is down. AppUpdate can't check for updates.");
             return null;
         } catch (IOException e) {

--- a/library/src/main/java/com/github/javiersantos/appupdater/RssParser.java
+++ b/library/src/main/java/com/github/javiersantos/appupdater/RssParser.java
@@ -9,8 +9,10 @@ import org.xml.sax.SAXException;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.ConnectException;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.URLConnection;
 import java.net.UnknownHostException;
 
 import javax.xml.parsers.ParserConfigurationException;
@@ -51,8 +53,10 @@ class RssParser {
 
     private InputStream getInputStream() {
         try {
-            return rssUrl.openConnection().getInputStream();
-        } catch (FileNotFoundException | UnknownHostException e) {
+            URLConnection connection = rssUrl.openConnection();
+            if (connection == null) return null;
+            return connection.getInputStream();
+        } catch (FileNotFoundException | UnknownHostException | ConnectException e) {
             Log.e("AppUpdater", "The XML updater file is invalid or is down. AppUpdate can't check for updates.");
             return null;
         } catch (IOException e) {

